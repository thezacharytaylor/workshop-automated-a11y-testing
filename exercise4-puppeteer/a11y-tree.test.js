/**
/* @jest-environment puppeteer
**/

import 'expect-puppeteer';
import path from 'path';
import fs from 'fs';

// describe('Accessibility Tree', () => {
//   beforeAll(async () => {
//     await page.goto('http://localhost:1234/passes');
//     // await page.goto('https://google.com');
//   });

//   it('should display heading text on page', async () => {
//     // await jestPuppeteer.debug();
//     // await expect(page).toMatch('Plan');
//     // await expect(page).toClick('button', { text: 'Join Now' });
//     await expect(page).toMatch('div.page-header-content', { text: 'Plan' });
//     // await expect(page).toMatch('body', { text: 'Google' });
//   });
// });

describe('Google', () => {
  beforeAll(async () => {
    await page.goto('https://google.com');
  });

  it('should display "google" text on page', async () => {
    await expect(page).toMatch('google');
  });
});
