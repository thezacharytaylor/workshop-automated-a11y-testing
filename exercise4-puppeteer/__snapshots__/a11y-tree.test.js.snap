// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accessibility Tree has the correct axTree 1`] = `
Object {
  "children": Array [
    Object {
      "level": 1,
      "name": "CampSpots",
      "role": "heading",
    },
    Object {
      "name": "CampSpots",
      "role": "link",
    },
    Object {
      "level": 2,
      "name": "Plan Your Trip",
      "role": "heading",
    },
    Object {
      "level": 2,
      "name": "Ways to Stay",
      "role": "heading",
    },
    Object {
      "level": 2,
      "name": "Resources",
      "role": "heading",
    },
    Object {
      "name": "Login",
      "role": "link",
    },
    Object {
      "children": Array [
        Object {
          "level": 2,
          "name": "Pick a Plan and Start Your Adventure Today!",
          "role": "heading",
        },
        Object {
          "level": 3,
          "name": "CampSpots Basic",
          "role": "heading",
        },
        Object {
          "level": 4,
          "name": "$99/year",
          "role": "heading",
        },
        Object {
          "name": "Renewed annually",
          "role": "StaticText",
        },
        Object {
          "name": "Unlimited overnight stays with No Camping Fees",
          "role": "StaticText",
        },
        Object {
          "name": "100% money back guarantee",
          "role": "StaticText",
        },
        Object {
          "name": "Access to spa facilities & restaurants",
          "role": "StaticText",
        },
        Object {
          "name": "Join Now",
          "role": "button",
        },
        Object {
          "name": "Vitae maximus turpis tempor",
          "role": "StaticText",
        },
        Object {
          "name": "nec",
          "role": "StaticText",
        },
        Object {
          "name": ".",
          "role": "StaticText",
        },
        Object {
          "level": 3,
          "name": "CampSpots Pro",
          "role": "heading",
        },
        Object {
          "level": 4,
          "name": "$139/year",
          "role": "heading",
        },
        Object {
          "name": "Renewed annually",
          "role": "StaticText",
        },
        Object {
          "name": "Unlimited overnight stays with No Camping Fees",
          "role": "StaticText",
        },
        Object {
          "name": "100% money back guarantee",
          "role": "StaticText",
        },
        Object {
          "name": "Access to spa facilities & restaurants",
          "role": "StaticText",
        },
        Object {
          "name": "Join Now",
          "role": "button",
        },
        Object {
          "name": "Lorem ipsum dolor",
          "role": "StaticText",
        },
        Object {
          "name": "sit",
          "role": "StaticText",
        },
        Object {
          "name": "amet, consectetur adipiscing elit.",
          "role": "StaticText",
        },
      ],
      "name": "Pick a Plan and Start Your Adventure Today!",
      "role": "generic",
    },
  ],
  "name": "CampSpots",
  "role": "RootWebArea",
}
`;
